[feature][needs-docs] Allow gpkg vector rename from browser

elpaso opened the pull request:
![gpkg-rename](https://user-images.githubusercontent.com/142164/48788486-36766a80-eceb-11e8-9ffe-d8b6000f7df5.gif)

elpaso committed:
[feature][needs-docs] Allow gpkg vector rename from browser

nyalldawson commented:
Looks good! My one suggestion would be to use QgsNewNameDialog instead of the raw text input here. It has nice handling for warning users if the name entered clashes with an existing one.

elpaso committed:
Support F2 for layer rename action

Also refactored some internal methods to
avoid duplication of code and use parent
class virtual rename() instead of a new
method for gpkg only.
